[SECRET-31] Add Secret Store documentation to dev-center

ZachEddy opened the pull request:
This PR adds documentation to the `dev-center` for Secret Store-related functionality. Secret Store is a new feature that lets users store passwords, credentials, etc. on Algorithmia so they don't have to hardcode sensitive data in their source code.

Commit:
Add Secret Store documentation to dev-center

bnlb commented on _pages/platform/algorithm-secrets.md:
```diff
@@ -0,0 +1,51 @@
+
+After your secret is configured, it can be updated or deleted on the settings page. Clicking "Update" will direct you to a form that is identical to the form shown above, and you'll be able to change the secret's name, description, and value.
+
+![Image of UI settings page for creating secrets](/developers/images/post_images/algorithm_secrets/settings_page_with_secret.png)
```
This alt tag is identical to a previous one. I'd update to something that reflects that a secret exists on the page now.
```suggestion
![Image of UI settings page with an existing secret](/developers/images/post_images/algorithm_secrets/settings_page_with_secret.png)
```

bnlb commented on _pages/platform/algorithm-secrets.md:
```diff
@@ -0,0 +1,51 @@
+
+Start by clicking "New secret" to create an algorithm secret. Next, give your secret a name, description, key, and value. The key corresponds to the environment variable name and the value is the secret itself.
+
+It can take up to one minute for new secrets to show up inside your algorithm. Similarly, when you update a secret, it can take one minute for that change to take effect.
```
Smart addition

bnlb reviewed (dismissed):
🔐

ZachEddy commented:
Thanks for the quick review, @bnlb! I updated the `alt-tag` like you mentioned. I met with @algorithmiaio/product on Friday and they requested I callout enterprise support more explicitly, so I added the `External secret stores` heading.

stevielkim reviewed (dismissed):
Awesome job on the docs!

Commit:
Update docs with recently-introduced web UI changes

ZachEddy commented:
Thank you, @StephanieKim! I was just about to merge this pull request, but I remembered there were some recent secret store UI changes. I updated one of the `.png` images in 603a6ab to reflect the new UI.

I dismissed your review by pushing changes, so would you mind re-approving?
